applications/serial_lte_modem/doc/TCPIP_AT_commands.rst

```diff
@@ -0,0 +1,1541 @@
+  * IPPROTO_TLS_1_2(258)
+  * IPPROTO_DTLS_1_2(273)
+
+Unsolicited notification
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <error> closed
+
+The ``<error>`` value is a negative integer, and it represents the error value according to the standard POSIX errorno.
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XSOCKET=1,1,0
+   #XSOCKET: 3, 6, 0
+   OK
+   AT#XSOCKET=1,2,0
+   #XSOCKET: 3, 17, 0
+   OK
+   AT#XSOCKET=0
+   #XSOCKET: 0, closed
+   OK
+   at#xsocket=1,1,0,16842755
+   #XSOCKET: 2, 1, 0, 258
+   OK
+   at#xsocket=1,2,0,16842755
+   #XSOCKET: 2, 2, 0, 273
+   OK
+
+Read command
+------------
+
+The read command allows you to check the socket handle.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <handle>[,<protocol>, <role>]
+
+* The ``<handle>`` value is an integer and it can be interpreted as follows:
+
+  * Positive - The socket is valid.
+  * ``0`` - The socket is closed.
+
+* The ``<protocol>`` value is present only in the response to a request to open the socket.
+  It can be one of the following:
+
+  * IPPROTO_TCP(6)
+  * IPPROTO_UDP(17)
+  * IPPROTO_TLS_1_2(258)
+  * IPPROTO_DTLS_1_2(273)
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
+
+Examples
+~~~~~~~~
+
+::
+
+   AT#XSOCKET?
+   #XSOCKET: 3, 6, 0
+   OK
+
+::
+
+   AT#XSOCKET?
+   #XSOCKET: 3, 17, 0
+   OK
+
+::
+
+   at#xsocket?
+   #XSOCKET: 2, 258, 0
+   OK
+
+::
+
+   at#xsocket?
+   #XSOCKET: 2, 273, 0
+   OK
+
+Test command
+------------
+
+The test command tests the existence of the command and provides information about the type of its subparameters.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET=?
+
+Response syntax
+~~~~~~~~~~~~~~~
+
+::
+
+   #XSOCKET: <list of op value>,<list of type value>,<list of roles>,<sec-tag>
+
+* The ``<role>`` parameter can accept one the following values:
+
+  * ``0`` - Client
+  * ``1`` - Server
+
+* The ``<type>`` parameter value depends on the presence of the <sec_tag> parameter.
+  When the ``<sec_tag>`` is not specified:
+
+  * 1: SOCK_STREAM for TCP
+  * 2: SOCK_DGRAM for UDP
+
+  When the ``<sec_tag>`` is specified:
+
+  * 1: SOCK_STREAM for TLS
+  * 2: SOCK_DGRAM for DTLS
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Open
+  * ``1`` - Close
+
+Examples
+~~~~~~~~
+
+::
+
+   at#xsocket=?
+   #XSOCKET: (0, 1), (1, 2),<sec_tag>
+   OK
+
+BSD socket options #XSOCKETOPT
+==============================
+
+The ``#XSOCKETOPT`` command allows you to get and set socket options.
+
+Set command
+-----------
+
+The set command allows you to get and set socket options.
+
+Syntax
+~~~~~~
+
+::
+
+   #XSOCKET=<op>,<name>[,<value>]
+
+* The ``<op>`` parameter can accept one the following values:
+
+  * ``0`` - Get
+  * ``1`` - Set
+
+For supported SET ``<names>``, refer to the `SETSOCKETOPT Service Spec Reference`_.
+``SO_RCVTIMEO(20)``, the ``<value>`` parameter is the *Receive Timeout* in seconds.
```

Comment:
is this description actually correct? Could do specify what ``SO_RCVTIMEO(20)`` and ``SO_ERROR(4)`` are?

Reply:
https://github.com/nrfconnect/sdk-zephyr/blob/master/include/net/socket.h#L798
They are BSDSocket standard socket options